builtin/credential/aws-ec2/path_login.go

```diff
@@ -288,6 +295,15 @@ func (b *backend) pathLoginUpdate(
 		}
 	}
 
+	// If the clientNonce is empty at this point, it means that its a first
+	// time login and that a nonce is not supplied. Create a random nonce
+	// to be associated for the instance ID.
+	if clientNonce == "" {
```

Comment:
Rather than the `if storedIdentity.DisallowReauthentication` check below, can we check whether reauthentication is allowed here (I would think in the role?) and if not, clear the nonce here (rather than set it and then unset it later), and if so, generate one if it's not supplied?